Xamarin.Forms.Platform.WPF/Renderers/MediaElementRenderer.cs

```diff
@@ -0,0 +1,299 @@
+				Control.UnloadedBehavior = MediaState.Close;
+				Control.Stretch = Element.Aspect.ToStretch();
+
+				Control.BufferingStarted += ControlBufferingStarted;
+				Control.BufferingEnded += ControlBufferingEnded;
+				Control.MediaOpened += ControlMediaOpened;
+				Control.MediaEnded += ControlMediaEnded;
+				Control.MediaFailed += ControlMediaFailed;
+
+				Element.SeekRequested += ElementSeekRequested;
+				Element.StateRequested += ElementStateRequested;
+				Element.PositionRequested += ElementPositionRequested;
+				UpdateSource();
+			}
+		}
+
+		void ElementPositionRequested(object sender, EventArgs e)
+		{
+			if (!(Control is null))
+			{
+				Controller.Position = Control.Position;
+			}
+		}
+		
+		void ElementStateRequested(object sender, StateRequested e)
+		{
+			_requestedState = e.State;
+
+			switch (e.State)
+			{
+				case MediaElementState.Playing:
+					if (Element.KeepScreenOn)
+					{
+						DisplayRequestActive();
+					}
+
+					Control.Play();
+					Controller.CurrentState = _requestedState;
+					break;
+
+				case MediaElementState.Paused:
+					if (Control.CanPause)
+					{
+						if (Element.KeepScreenOn)
+						{
+							DisplayRequestRelease();
+						}
+
+						Control.Pause();
+						Controller.CurrentState = _requestedState;
+					}
+					break;
+
+				case MediaElementState.Stopped:
+					if (Element.KeepScreenOn)
+					{
+						DisplayRequestRelease();
+					}
+
+					Control.Stop();
+					Controller.CurrentState = _requestedState;
+					break;
+			}
+
+			Controller.Position = Control.Position;
+		}
+
+		void ElementSeekRequested(object sender, SeekRequested e)
+		{
+			Control.Position = e.Position;
+			Controller.Position = Control.Position;
+		}
+
+		void UpdateSource()
+		{
+			if (Element.Source is null)
+				return;
+
+			if (Control.Clock != null)
+				Control.Clock = null;
+
+			if (Element.Source is UriMediaSource uriSource)
+			{
+				if (uriSource.Uri.Scheme == "ms-appx")
+				{
+					Control.Source = new Uri(uriSource.Uri.ToString().Replace("ms-appx://", "pack://application:,,,"));
+				}
+				else if (uriSource.Uri.Scheme == "ms-appdata")
+				{
+					string filePath = string.Empty;
+
+					if (uriSource.Uri.LocalPath.StartsWith("/local"))
+					{
+						// WPF doesn't have the concept of an app package local folder so using My Documents as a placeholder
+						filePath = System.IO.Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.MyDocuments), uriSource.Uri.LocalPath.Substring(7));
+					}
+					else if (uriSource.Uri.LocalPath.StartsWith("/temp"))
+					{
+						filePath = System.IO.Path.Combine(System.IO.Path.GetTempPath(), uriSource.Uri.LocalPath.Substring(6));
+					}
+					else
+					{
+						throw new ArgumentException("Invalid Uri", "Source");
+					}
+
+					Control.Source = new Uri(filePath);
+				}
+				else if (uriSource.Uri.Scheme == "https")
+				{
+					throw new ArgumentException("HTTPS Not supported on WPF", "Source");
```

Comment:
Maybe we should mention that HTTP is so that people don't think ALL web sources are bad.

Reply:
I've added more detail to the message. The documentation will need to call this out too.